src/solids4FoamModels/fluidModels/pimpleFluid/pimpleFluid.foamextend.C

```diff
@@ -138,6 +138,13 @@ pimpleFluid::pimpleFluid
 
     setRefCell(p(), pimple().dict(), pRefCell_, pRefValue_);
     mesh().schemesDict().setFluxRequired(p().name());
+
+    if (useFoamExtend40formulation_ && robin_.patchIDs().size() > 0)
+    {
+        FatalErrorIn(type())
+            << "Currently, the Robin coupling approach only works with "
+            << "useFoamExtend40formulation turned off" << abort(FatalError);
```

Comment:
From this message, I am not sure whether the Robin BCs do not work only with the formulation of FE40 or when it is compiled with FE40. I remember using the Robin conditions with FE40 too, so I believe they work fine with FE40, actually, unless you did any other modification that I am not aware of. Also, in the pEqn file for FE40 the modifications to account for those BCs are included and should work.